pageserver/src/tenant/timeline/detach_ancestor.rs

```diff
@@ -1156,12 +1162,16 @@ fn check_no_archived_children_of_ancestor(
     detached: &Arc<Timeline>,
     ancestor: &Arc<Timeline>,
     ancestor_lsn: Lsn,
+    detach_behavior: DetachBehavior,
 ) -> Result<(), Error> {
     let timelines = tenant.timelines.lock().unwrap();
     let timelines_offloaded = tenant.timelines_offloaded.lock().unwrap();
-    for timeline in reparentable_timelines(timelines.values(), detached, ancestor, ancestor_lsn) {
-        if timeline.is_archived() == Some(true) {
-            return Err(Error::Archived(timeline.timeline_id));
+    if let DetachBehavior::NoAncestorAndReparent = detach_behavior {
```

Comment:
I think a match would be better here, that way if someone adds something to `DetachBehaviour`, they compiler points out this location.